### DISCUSSION TITLE
How to type hint in this situation

### DISCUSSION THREAD

**Discussion by Harry-Jing** [pydantic/pydantic#5211] [Question]
For example, I want to build a `Model` that have a `word_list` witch store value type is `list[str]`. But user can input either `list[str]` or `str`. The `Model` can auto trans into `list[str]`.

The only way I know is to write just like below
```python
from pydantic import BaseModel, validator


class Model(BaseModel):
    word_list: list[str]

    @validator('word_list', pre=True)
    def word_list_validator(cls, v):
        return list(v)

```

However that can't give type hint when user input

**Comment by Harry-Jing** [comment]
![image](https://user-images.githubusercontent.com/69247286/225916002-5b6caa8f-cae8-4560-9924-9417b55b509f.png)

**Comment by FossenWang** [comment]
https://docs.pydantic.dev/usage/types/#custom-data-types
add a validator to  `__get_validators__`
